### PR TITLE
[avro-c] enable Snappy codec

### PR DESCRIPTION
**Pull request by mispecto**
Adds Snappy codec to Avro-C library.

### CODE REVIEW

File: ports/avro-c/snappy.patch [hunk 1]
```diff
@@ -0,0 +1,56 @@
+diff -ru b/c/src/codec.c a/lang/c/src/codec.c
+--- b/lang/c/src/codec.c	2020-01-23 16:18:15.119970300 +0200
++++ a/lang/c/src/codec.c	2020-01-23 19:31:41.679834300 +0200
+@@ -21,6 +21,9 @@
+ #  if defined(__APPLE__)
+ #    include <libkern/OSByteOrder.h>
+ #    define __bswap_32 OSSwapInt32
++#  elif defined(WIN32)
```
Comment (c72578) [inline ports/avro-c/snappy.patch]: The underscore is missing from the preprocessor macro. Should be:
`#  elif defined(_WIN32)`

File: ports/avro-c/portfile.cmake [hunk 2]
```diff
@@ -12,13 +12,13 @@ vcpkg_from_github(
         avro.patch
         avro-pr-217.patch
         fix-build-error.patch # Since jansson updated, use jansson::jansson instead of the macro ${JANSSON_LIBRARIES}
+        snappy.patch
```
Comment (c72578) [inline ports/avro-c/portfile.cmake]: Could you please add a comment with info, where this is reported or fixed upstream?

File: ports/avro-c/CONTROL [hunk 3]
```diff
@@ -2,4 +2,4 @@ Source: avro-c
 Version: 1.8.2-3
```
Comment (JackBoosY) [inline ports/avro-c/CONTROL]: Can you update version here? See [documentation](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#update-the-version-field-in-the-control-file-of-any-modified-ports).

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Comment by spektom** [comment]
@strega-nil Can you tell please whether [vcpkg-Linux-PR-test](https://dev.azure.com/vcpkg/c1ee48cb-0df2-4ab3-8384-b1df5a79fe53/_build/results?buildId=27363) test failures are related to this PR?

**Comment by JackBoosY** [comment]
Error logs:
```
/usr/include/pthread.h:399:12: error: ‘cpu_set_t’ does not name a type
      const cpu_set_t *__cpuset)
            ^~~~~~~~~
/usr/include/pthread.h:406:6: error: ‘cpu_set_t’ has not been declared
      cpu_set_t *__cpuset)
      ^~~~~~~~~
/usr/include/pthread.h:476:14: error: ‘cpu_set_t’ does not name a type
        const cpu_set_t *__cpuset)
              ^~~~~~~~~
/usr/include/pthread.h:481:8: error: ‘cpu_set_t’ has not been declared
        cpu_set_t *__cpuset)
```

**Comment by spektom** [comment]
> 
> /usr/include/pthread.h:481:8: error: ‘cpu_set_t’ has not been declared
>         cpu_set_t *__cpuset)
> ```

Can you point to the full log file? I don't see how this is related to the change.

**Comment by JackBoosY** [comment]
[port build failure logs.zip](https://github.com/microsoft/vcpkg/files/4127868/port.build.failure.logs.zip)

**Comment by spektom** [comment]
> 
> 
> [port build failure logs.zip](https://github.com/microsoft/vcpkg/files/4127868/port.build.failure.logs.zip)

These failures seem unrelated. Failing components are: `cctz`,  `clblas`,  `discord-rpc`,  `matroska`.
Is there a chance building environment was changed "from the outside"?

**Comment by JackBoosY** [comment]
@spektom Agreed.
@vicroms Can you have a look?

**Comment by vicroms** [comment]
Failures were due to unrelated issue on CI that has now been fixed.

Thanks for the PR!

**Pull request merged**